src/sys/bout_types.cxx

```diff
@@ -20,6 +20,7 @@ const std::string& DIFF_METHOD_STRING(DIFF_METHOD location) {
 }
 
 const std::string& REGION_STRING(REGION region) {
+  ASSERT2(region >= 0 && region <= 4);
```

Comment:
I think this could go below the `REGIONtoString` declaration below and could then read `ASSERT2(region >=0 && region <= REGIONtoString.size())`. Alternatively we could do `ASSERT2(REGIONtoString.count(region) == 1)` to ask the map if it can find the requested region or not -- this then doesn't assume anything about the `enum` numbering etc.

Reply:
Cleanest is probably to change the `at` to `find`, and throw `BoutException` if not found

Reply:
Is REGION meant to stay? Or will it be deprecated soon?

Reply:
We're planning on deprecating `REGION` in the next release.

Reply:
I have changed to at - in all cases, to be consistent. To avoid code duplication I wrote a template, that has the small disadvantage of not saying what enum we are trying to convert, but I think that should be fine, as the backtrace should make that clear ...